tst/EnergyPlus/unit/OutputReportTabular.unit.cc

```diff
@@ -9610,3 +9611,98 @@ TEST_F(SQLiteFixture, ORT_EndUseBySubcategorySQL_DualUnits)
+
+    // We enable the reports we care about, making sure we have the right ones
+    EXPECT_EQ("HVACSizingSummary", state->dataOutRptPredefined->reportName(6).name);
+    state->dataOutRptPredefined->reportName(6).show = true;
+
+    OutputReportTabular::OpenOutputTabularFile(*state);
+
+    WritePredefinedTables(*state);
+
+    OutputReportTabular::CloseOutputTabularFile(*state);
+
+    std::vector<std::string> lines = read_lines_in_file(DataStringGlobals::outputTblHtmFileName);
+
+    // Lambda helper to locate a line in the html file, and compare that line with the expected html after trimming
+    auto compare_html_output = [this, &lines](const std::string& lookup, const std::string& expectedHTMLString) {
+        std::string found_cell;
+        for (const auto& line: lines) {
+            if (line.find(lookup) != std::string::npos) {
+                found_cell = line;
+                break;
+            }
+        }
+        EXPECT_FALSE(found_cell.empty())
+            << "Did not find the lookup string '" << lookup
+            << "' string in the html output at '" << DataStringGlobals::outputTblHtmFileName
+            << "'..." << '\n' << delimited_string(lines);
+
+        // Trim leading and trailing spaces
+        found_cell.erase(0, found_cell.find_first_not_of(' ')); // ltrim
+        found_cell.erase(found_cell.find_last_not_of(' ') + 1); // rtrim
+
+        EXPECT_EQ(expectedHTMLString, found_cell) << found_cell;
+    };
+
+    compare_html_output("My Coil", "<td align=\"right\">My Coil &lt;coil is DX&gt;</td>");
+
+    // Note that I DO NOT expect `'` to be escaped by `&apos;` like it would in xml. Technically HTML4 doesn't support that, though most browsers
+    // would anyways. Also, escaping single and double quotes is only needed inside attributes
+    compare_html_output("My Design Day", "<td align=\"right\">My Design Day where it's &gt;= 8&deg;</td>");
+
+
+    // We ensure that SQL doesn't have the same escape
+    for (const std::string reportName: {"HVACSizingSummary"}) {
+
+
+        auto result = queryResult("SELECT RowName, Value From TabularDataWithStrings "
+                                 "WHERE ReportName = \"" + reportName + "\""
+                                 "  AND ColumnName = \"Coil Leaving Air Humidity Ratio at Ideal Loads Peak\"",
+                                 "TabularDataWithStrings");
+
+        EnergyPlus::sqlite->sqliteCommit();
+
+        EXPECT_EQ(1u, result.size());
+        // Because the table has 8 cols
+        EXPECT_EQ(8u, result[0].size());
+
+        // 0.006 is a ratio, so unitconv = 1
+        std::string s = result[0][0];
+        // Trim the string, it has leading spaces
+        //s.erase(std::remove_if(s.begin(), s.end(), ::isspace), s.end());
+
+        EXPECT_EQ("My Coil <coil is DX>", s);
+
+        EXPECT_EQ("My Design Day where it's >= 8\u00B0", result[0][1]);
+    }
```

Comment:
And we check that the HTML change didn't leak in the SQL file as well.

Reply:
Awesome!